src/assets/Standards/CollectibleStandards/ERC721/ERC721Standard.ts

```diff
@@ -221,46 +221,50 @@ export class ERC721Standard {
     name: string | undefined;
     image: string | undefined;
   }> => {
-    const [isERC721, supportsMetadata] = await Promise.all([
-      this.contractSupportsBase721Interface(address),
-      this.contractSupportsMetadataInterface(address),
-    ]);
-    let tokenURI, symbol, name, image;
-    if (supportsMetadata) {
-      [symbol, name] = await Promise.all([
-        this.getAssetSymbol(address),
-        this.getAssetName(address),
-      ]);
+    const isERC721 = await this.contractSupportsBase721Interface(address);
+    if (!isERC721) {
+      throw new Error("This isn't a valid ERC721 contract");
+    }
+
+    let tokenURI, image, symbol, name;
 
-      if (tokenId) {
+    // TODO upgrade to use Promise.allSettled for name/symbol when we can refactor to use es2020 in tsconfig
+    try {
+      symbol = await this.getAssetSymbol(address);
```

Comment:
What would we expect this to throw if the asset name was missing? It would be safer to encode the specific error expected so that we don't swallow unexpected errors. Same for the other empty catch blocks

Reply:
Something like "Execution reverted", but perhaps also other flavors if the type of the return value doesn't match etc. Possibly many different combinations.

Reply:
Ah, I hadn't realized this was a network call. This could differ between networks then I guess? And it's via `web3.eth.contract` which I would not expect to have well-defined error return types...Hmm.

Reply:
Fair enough, we can revisit this when we're replacing web3. No sense in trying to understand web3's error handling.

Reply:
I learned recently (https://github.com/MetaMask/controllers/pull/830/) that it doesn't do any validation of the returned data types. It just tries to parse them as the expected data type and throws a random error if that fails because the data type is wrong. So you'll get errors like "Can't construct new BigNumber() with non-hex input".